scriptmodules/supplementary/wifi.sh

```diff
@@ -0,0 +1,114 @@
+    choice=$("${cmd[@]}" 2>&1 >/dev/tty)
+    if [[ -n "$choice" ]]; then
+        psk="$choice"
+    fi
+
+    echo -e 'auto lo\n\niface lo inet loopback\niface eth0 inet dhcp\n\nallow-hotplug wlan0\nauto wlan0\n\niface wlan0 inet manual\nwpa-roam /etc/wpa_supplicant/wpa_supplicant.conf\n\niface default inet dhcp' > "/etc/network/interfaces"
+
+    echo -e 'ctrl_interface=DIR=/var/run/wpa_supplicant GROUP=netdev\nupdate_config=1\n\nnetwork={\n\tssid="'$ssid'"\n\tpsk="'$psk'"\n}' > "/etc/wpa_supplicant/wpa_supplicant.conf"
+    ifdown wlan0
+    ifup wlan0
+    printMsgs "dialog" "You are now connected to: $ssid\n\nConfigurations have been saved to /etc/network/interfaces and /etc/wpa_supplicant/wpa_supplicant.conf"
+}
+
+function wifi_wep() {
+    ifconfig wlan0 up
+    cmd=(dialog --backtitle "$__backtitle" --inputbox "Please Enter the SSID of the Network You Would Like to Connect to" 10 60 "$ssid")
+    choice=$("${cmd[@]}" 2>&1 >/dev/tty)
+    if [[ -n "$choice" ]]; then
+        ssid="$choice"
+    fi
+
+    cmd=(dialog --backtitle "$__backtitle" --insecure --passwordbox "Please Enter Your WEP Key" 10 60 $psk)
+    choice=$("${cmd[@]}" 2>&1 >/dev/tty)
+    if [[ -n "$choice" ]]; then
+        psk="$choice"
+    fi
+
+    echo -e 'auto lo\n\niface lo inet loopback\niface eth0 inet dhcp\n\nallow-hotplug wlan0\nauto wlan0\n\niface wlan0 inet manual\nwpa-roam /etc/wpa_supplicant/wpa_supplicant.conf\n\niface default inet dhcp' > "/etc/network/interfaces"
+
+    echo -e 'ctrl_interface=DIR=/var/run/wpa_supplicant GROUP=netdev\nupdate_config=1\n\nnetwork={\n\tssid="'$ssid'"\n\tkey_mgmt=NONE\n\twep_tx_keyidx=0\n\tpsk='"$psk"'\n}' > "/etc/wpa_supplicant/wpa_supplicant.conf"
+    ifdown wlan0
+    ifup wlan0
+    printMsgs "dialog" "You are now connected to: $ssid\n\nConfigurations have been saved to /etc/network/interfaces and /etc/wpa_supplicant/wpa_supplicant.conf"
+}
+
+function wifi_open() {
+    ifconfig wlan0 up
+    cmd=(dialog --backtitle "$__backtitle" --inputbox "Please Enter the SSID of the Network You Would Like to Connect to" 10 60 "$ssid")
+    choice=$("${cmd[@]}" 2>&1 >/dev/tty)
+    if [[ -n "$choice" ]]; then
+        ssid="$choice"
+    fi
+
+    echo -e 'auto lo\n\niface lo inet loopback\niface eth0 inet dhcp\n\nallow-hotplug wlan0\nauto wlan0\n\niface wlan0 inet manual\nwpa-roam /etc/wpa_supplicant/wpa_supplicant.conf\n\niface default inet dhcp' > "/etc/network/interfaces"
+
+    echo -e 'ctrl_interface=DIR=/var/run/wpa_supplicant GROUP=netdev\nupdate_config=1\n\nnetwork={\n\tssid="'$ssid'"\n\tkey_mgmt=NONE\n}' > "/etc/wpa_supplicant/wpa_supplicant.conf"
+    ifdown wlan0
+    ifup wlan0
+    printMsgs "dialog" "You are now connected to: $ssid\n\nConfigurations have been saved to /etc/network/interfaces and /etc/wpa_supplicant/wpa_supplicant.conf"
+}
+
+function configure_wifi() {
+
+    local ip_int=$(ifconfig | grep -Eo 'inet (addr:)?([0-9]*\.){3}[0-9]*' | grep -Eo '([0-9]*\.){3}[0-9]*' | grep -v '127.0.0.1')
+
```

Comment:
This will likely fail on a non English install as eg german ifconfig has "inet Adresse"

I spent some time looking into this before, and the best solution I came up with was

```
ip route get 8.8.8.8 2>/dev/null | head -1 | cut -d' ' -f8
```

which is what is used in the bashwelcometweak.